src/Symfony/Bridge/Doctrine/DataCollector/DoctrineDataCollector.php

```diff
@@ -180,32 +220,33 @@ private function sanitizeQuery(string $connectionName, array $query): array
      * indicating if the original value was kept (allowing to use the sanitized
      * value to explain the query).
      */
-    private function sanitizeParam($var): array
+    private function sanitizeParam($var, ?\Throwable $error): array
     {
         if (\is_object($var)) {
-            $className = \get_class($var);
+            return [new ObjectParameter($var, $stringable = method_exists($var, '__toString'), $error), false, $stringable && !$error];
+        }
 
-            return method_exists($var, '__toString') ?
-                [sprintf('/* Object(%s): */"%s"', $className, $var->__toString()), false] :
-                [sprintf('/* Object(%s) */', $className), false];
+        if ($error) {
+            return ['⚠ '.$error->getMessage(), false, false];
         }
 
         if (\is_array($var)) {
             $a = [];
-            $original = true;
+            $explainable = $runnable = true;
             foreach ($var as $k => $v) {
-                list($value, $orig) = $this->sanitizeParam($v);
-                $original = $original && $orig;
+                list($value, $e, $r) = $this->sanitizeParam($v, null);
+                $explainable = $explainable && $e;
+                $runnable = $runnable && $r;
                 $a[$k] = $value;
             }
 
-            return [$a, $original];
+            return [$a, $explainable, $runnable];
         }
 
         if (\is_resource($var)) {
-            return [sprintf('/* Resource(%s) */', get_resource_type($var)), false];
+            return [sprintf('/* Resource(%s) */', get_resource_type($var)), false, false];
```

Comment:
should we rather dump the resource with VarDumper now that params are cloned ?

Reply:
We can probably improve it too but in a further PR.

Reply:
Let's do it here if it's changing only one line.

Reply:
not a one-liner, let's consider in another PR then :)